packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx

```diff
@@ -466,6 +474,26 @@ const QueryBuilderPostFilterConditionEditor = observer(
         node.condition.operator.getDefaultFilterConditionValue(node.condition),
       );
     };
+    const debouncedTypeaheadSearch = useMemo(
```

Comment:
I want to simplify the API of `BasicValueSpecification` further here. Also, I just thought of a case I don't know if we need to handle, that is what happens if we do typeahead search and the returns are `booleans` or `numbers`?

Reply:
we can further enhance that for numbers.
Maybe not yet for booleans.